media-player/onju-voice.yaml

```diff
@@ -0,0 +1,482 @@
+  high_voltage_reference: 2.4V
+  
+  filter_mode: IIR_16
+  debounce_count: 2
+  noise_threshold: 0
+  jitter_step: 0
+  smooth_mode: IIR_2
+
+  denoise_grade: BIT8
+  denoise_cap_level: L0
+
+binary_sensor:
+  - platform: esp32_touch
+    id: volume_down
+    pin: GPIO4
+    threshold: 539000 # will automatically be tuned at runtime
+    on_press: 
+      then:
+        - light.turn_on: left_led
+        - script.execute:
+            id: set_volume
+            volume: -0.05
+        - delay: 1s
+        - while:
+            condition:
+              binary_sensor.is_on: volume_down
+            then:
+              - script.execute:
+                  id: set_volume
+                  volume: -0.05
+              - delay: 150ms
+    on_release: 
+      then:
+        - light.turn_off: left_led
+        
+  - platform: esp32_touch
+    id: volume_up
+    pin: GPIO2
+    threshold: 580000 # will automatically be tuned at runtime
+    on_press: 
+      then:
+        - light.turn_on: right_led
+        - script.execute:
+            id: set_volume
+            volume: 0.05
+        - delay: 1s
+        - while:
+            condition:
+              binary_sensor.is_on: volume_up
+            then:
+              - script.execute:
+                  id: set_volume
+                  volume: 0.05
+              - delay: 150ms
+    on_release: 
+      then:
+        - light.turn_off: right_led
+        
+  - platform: esp32_touch
+    id: action
+    pin: GPIO3
+    threshold: 751000 # will automatically be tuned at runtime
+    on_click:
+      - if:
+          condition:
+            or:
+              - switch.is_off: use_wake_word
+              - binary_sensor.is_on: mute_switch
+          then:
+            - if:
+                condition: voice_assistant.is_running
+                then:
+                  - voice_assistant.stop:
+                  - script.execute: reset_led
+                else:
+                  - voice_assistant.start:
+          else:
+            - voice_assistant.stop
+            - delay: 1s
+            - script.execute: reset_led
+            - script.wait: reset_led
+            - voice_assistant.start_continuous:
+            
+  - platform: gpio
+    id: mute_switch
+    pin:
+      number: GPIO38
+      mode: INPUT_PULLUP
+    name: Disable wake word
+    on_press:
+      - script.execute: turn_on_wake_word
+    on_release:
+      - script.execute: turn_off_wake_word
+      
+  - platform: status
+    id: api_connection
+    filters:
+      - delayed_on: 1s
+    on_press:
+      - if:
+          condition:
+            and:
+              - switch.is_on: use_wake_word
+              - binary_sensor.is_off: mute_switch
+          then:
+            - voice_assistant.start_continuous:
+    on_release:
+      - if:
+          condition:
+            and:
+              - switch.is_on: use_wake_word
+              - binary_sensor.is_off: mute_switch
+          then:
+            - voice_assistant.stop:
```

Comment:
ESPHome 2023.10.5 has some new triggers on the voice assistant that will provide a better experience for this. See https://github.com/esphome/firmware/pull/86